src/wasm/wasm-binary.cpp

```diff
@@ -1567,6 +1573,12 @@ void WasmBinaryWriter::writeType(Type type) {
         type = Type(type.getHeapType().getTop(), Nullable);
       }
     }
+    // If the type is exact, emit the exact prefix and continue on without
+    // considering exactness.
+    if (type.isExact()) {
+      o << S32LEB(BinaryConsts::EncodedType::exact);
+      type = Type(type.getHeapType(), type.getNullability(), Inexact);
```

Comment:
Where below do we need this line for? (1579 makes sense to me, but 1580 I don't see yet)

Reply:
It looks like it's not necessary in the current code, but if we refactored the code below to check equality with specific reference types, for example, it would be important that we ignore exactness from this point.

Reply:
Fair enough, I guess it's safer for the future.